runtime/lua/vim/filetype.lua

```diff
@@ -1312,6 +1319,7 @@ local pattern = {
   ["tmac%..*"] = starsetf('nroff'),
   ["zlog.*"] = starsetf('zsh'),
   ["zsh.*"] = starsetf('zsh'),
+  ["ae%d+%.txt"] = 'mail',
```

Comment:
This is going to be a good test case for the global priority implementation (`help`>`mail`>`txt`).

Reply:
I checked locally that this all works out ;)

Reply:
Wait, no, I forgot a case. Patterns come before extensions, so a file `ae3.txt` is correctly classified as `mail`, but putting the right modeline in there won't make it a help file.

Reply:
No, that's not how this works -- `helpfile`s are not matched by modeline (that's a purely negative marker) but by location:
```viml
" Vim help file
au BufNewFile,BufRead $VIMRUNTIME/doc/*.txt setf help
```

Reply:
There's a duplication here, see https://github.com/neovim/neovim/blob/master/runtime/filetype.vim#L2404

Reply:
Yes I noticed that too, but it's in upstream Vim too so it's Not Our Problem™️

Reply:
It's not a duplication; the order in Vim is (likely) intentional to have the right behavior (make `txt` files in those directories `help`, unless they're not help files). 

We should instead use explicit conditionals or priorities.

Reply:
Note that all txt files in `$VIMRUNTIME/doc` do indeed have that modeline and are correctly classified, so right no the pattern isn't needed. That leaves the case of a `ae%d+%.txt` file containing a help modeline, but I'm not exactly sure it matters (then again, I'm not sure why we need to check the modeline anyways, shouldn't that be run after all builtin code anyways?).

Reply:
No, it's run _before_ (and, eventually, instead -- that's the whole point of this exercise ;))

Reply:
By the way, there _is_ a duplication:
```viml
" Mail (for Elm, trn, mutt, muttng, rn, slrn, neomutt)
au BufNewFile,BufRead snd.\d\+,.letter,.letter.\d\+,.followup,.article,.article.\d\+,pico.\d\+,mutt{ng,}-*-\w\+,mutt[[:alnum:]_-]\\\{6\},neomutt-*-\w\+,neomutt[[:alnum:]_-]\\\{6\},ae\d\+.txt,/tmp/SLRN[0-9A-Z.]\+,*.eml setf mail
```
(line 1013 of `filetype.vim`)

Reply:
Ok, I wasn't aware of any connection of this to modelines. Nevermind, though, just let me know if I should add a function for `ae%s+%.txt` instead that also checks the modeline.

Reply:
No, we want to replicate exactly the behavior of `filetype.vim`, and there's no modeline check there, either.

Reply:
I tried adding `$VIMRUNTIME/doc/*.txt` as help files, but it seems to mee that's a pretty brittle thing. To test, I passed `VIMRUNTIME=./runtime`, and that path is taken verbatim into the `$VIMRUNTIME` variable. As per the docs, the aucmd checks against the short file name and the full path, none of which contain `./runtime`. I'm not sure how that is resolved in filetype.lua, but maybe it suffers the same problem.

Reply:
Yeah, that's a separate issue. No need to address this in this PR. (The doc files all contain modelines, so that is just to avoid setting the filetype twice.)